Add Two New User Scripts

Rayman1103 opened the pull request:
- Add user_script_locals_yellow_team, which makes the natives on a planet use their yellow color for units and CPs.
- Add user_script_rhenvar2_cp_fix, which fixes the issue where CPs 1 and 4 aren't counted as valid CPs for Conquest mode.

Rayman1103 committed:
Add Two New User Scripts

- Add user_script_locals_yellow_team, which makes the natives on a planet use their yellow color for units and CPs.
- Add user_script_rhenvar2_cp_fix, which fixes the issue where CPs 1 and 4 aren't counted as valid CPs for Conquest mode.

BAD-AL commented on src/scripts/user_script_rhenvar2_cp_fix.lua:
```diff
@@ -0,0 +1,29 @@
+--attempt to take control of (or listen to the calls of) the ScriptPostLoad function
+if ScriptPostLoad and not RhenVar2CPFix_ScriptPostLoad then
+	--backup the current ScriptPostLoad function
+	RhenVar2CPFix_ScriptPostLoad = ScriptPostLoad
```
any reason for this not to be local (RhenVar2CPFix_ScriptPostLoad)?

Rayman1103 replied:
I guess I never gave it a thought. Is local better in this situation? If so, I can update this and the previous Hero user_scripts and change those calls to local as well.

BAD-AL replied:
If it is intended that it be acted upon outside this script, then it should be global; otherwise local.

Rayman1103 replied:
Alright, I'll update them later and make them local. Thanks for the advice.

BAD-AL commented on src/scripts/user_script_rhenvar2_cp_fix.lua:
```diff
@@ -0,0 +1,29 @@
+		-- let the original function happen and catch the return value
+		local RhenVar2CPFix_SPLreturn = {RhenVar2CPFix_ScriptPostLoad(unpack(arg))}
+		
+		if __thisMapsCode__ == "rhn" and __thisMapsMode__ == "rhenvar2_conquest" then
```
Where do these come from?
```LUa
__thisMapsCode__
__thisMapsMode__
```
Please leave a reference on how these were populated with data.

Rayman1103 replied:
I saw they were defined in **utility_functions2.lua**, and set within the **ReadDataFile** override in the same file. The **fakeconsole_functions.lua** file already used them for level-specific commands.

BAD-AL replied:
Ok, I see where they are from (That code from utility_functions2.lua looks kinda buggy [assuming 3 letters for a folder always], but it's been in so long it's practically a tradition).
Please leave a reference in this code that they are set from utility_functions2.lua and show an example of what they will be when called like 
```LUA
ReadDataFile("dc:ABC\\abc.lvl", "abc_conquest")
ReadDataFile("dc:DEF\\def.lvl", "def_con")
```

Rayman1103 replied:
Since those variables could potentially be unreliable, I just changed them now and made the ScriptPostLoad overrides local.

Rayman1103 committed:
Minor Tweaks

- Made the ScriptPostLoad override local.
- Changed the method used to detect if playing on Rhen Var. Removed the __thisMapsCode__ and __thisMapsMode__ variables.

BAD-AL commented on src/scripts/user_script_rhenvar2_cp_fix.lua:
```diff
@@ -6,14 +6,16 @@
 		
-		if __thisMapsCode__ == "rhn" and __thisMapsMode__ == "rhenvar2_conquest" then
+		--check if the map's Rhen Var Citadel and if the Conquest objective is defined
+		if GetWorldFilename() == "rhenvar2" and ObjectiveConquest ~= nil then
```
I didn't even know we had this function available in the game! :)
